### DISCUSSION TITLE
traefik ingress is stuck in progress in ArgoCD, address field is empty by default

### DISCUSSION THREAD

**Discussion by HendrikLevering** [mysticaltech/terraform-hcloud-kube-hetzner#1048] [Ideas]
If you use a Hetzner cluster with traefik and you deploy an  app with ingress via ArgoCD, then ArgoCD is stuck in progress. The reason is: ArgoCD will check the ingress address to field. It will only switch to healthy, if the field is not empty. In the default Hetzner / Traefik setup this field is empty.
You can make traefik populate this field by adding this to traefik_values:

additionalArguments:
  - "--providers.kubernetesingress.ingressendpoint.publishedservice=traefik/traefik"

This will populate the address field with the load balancer address.
Ref:
https://github.com/argoproj/argo-cd/issues/14607
https://community.traefik.io/t/traefik-ingress-not-getting-loadbalancer-ip/17445

However, I tried this with an existing running cluster. I added the additional argument. The result was, that a second load balancer was created and the first one was somehow in an orphaned state. So I had to manually delete the 2nd load balancer from Hetzner, delete the traefik service and create a new one, which reuses the old load balancer. (make a backup of the existing service before deletion!) 
Ref for loadbalancer reusage:
https://github.com/hetznercloud/hcloud-cloud-controller-manager/blob/main/docs/load_balancers.md#reference-existing-load-balancers
I created the new one by putting the service in a yaml file and apply it to the cluster.

I put this information here, so that others with this problem may find it.

I have one question:
Do I have to be worried that my new traefik service is overidden? I read that traefik is redeployed on k3 start. I tested this by restarting a node, but so far it seems to continue working fine

**Comment by HendrikLevering** [comment]
Ah and of course my proposal is to add
```
"--providers.kubernetesingress.ingressendpoint.publishedservice=traefik/traefik"
```
by default, so that ingress has always a valid address.

**Comment by Silvest89** [comment]
Yes, this should actually be the default. Else services of type Ingress will point to an empty loadbalancer.
This is also needed for external-dns to work. 

You can just use instead of traefik_values
`traefik_additional_options = ["--providers.kubernetesingress.ingressendpoint.publishedservice=traefik/traefik"]`

Or if you prefer the HelmChartConfig
```
apiVersion: helm.cattle.io/v1
kind: HelmChartConfig
metadata:
  name: traefik
  namespace: kube-system
spec:
  valuesContent: |-
    additionalArguments:
      - "--providers.kubernetesingress.ingressendpoint.publishedservice=traefik/traefik"
```

**Comment by Silvest89** [comment]
@mysticaltech Your opinion?

**Comment by mysticaltech** [comment]
@HendrikLevering @Silvest89 Yes, definitely best to add the field by default. PR most welcome 🙏

**Comment by Silvest89** [comment]
@mysticaltech https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/pull/1049
